src/main/java/com/example/employee/service/EmployeeService.java

```diff
@@ -15,27 +16,28 @@ public class EmployeeService {
     private final EmployeeRepository employeeRepository;
     private final EmployeeMapper employeeMapper;
 
-    public EmployeeDTO createAndUpdateEmployee(EmployeeDTO employeeDTO){
+    public EmployeeDTO createAndUpdateEmployee(EmployeeDTO employeeDTO) {
         Employee employee = employeeMapper.dtoToEntity(employeeDTO);
-        if (employeeRepository.findByEmail(employeeDTO.getEmail()) == null){
-        employeeRepository.save(employee);
-        return employeeMapper.entityToDto(employee);
+        if (employeeRepository.findByEmail(employeeDTO.getEmail()) == null) {
+            employeeRepository.save(employee);
+            return employeeMapper.entityToDto(employee);
         } else {
             return employeeDTO;
```

Comment:
Actually, here is a bit different logic, we want to create and save employees in db only in case if it already is not in db.

So, we are checking if employee by indicated email exists. 
Then if the employee is found we do not create new employee and return founded one, But if it is not found we create new one and save it.